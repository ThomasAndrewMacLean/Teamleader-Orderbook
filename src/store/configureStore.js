import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from './../reducers/reducers';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { routerMiddleware } from 'react-router-redux';

import createHistory from 'history/createBrowserHistory';



const logger = createLogger({
    collapsed: true
});

const history = createHistory();
const routerMw = routerMiddleware(history);

const composeEnhancers =
    typeof window === 'object' &&
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
            // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
        }) : compose;


export default function configureStore() {
    return createStore(

        rootReducer,
        composeEnhancers(
            applyMiddleware(routerMw, logger, thunk))
    );
}
