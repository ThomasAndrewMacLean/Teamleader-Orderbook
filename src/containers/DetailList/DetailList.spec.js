import React from 'react';
import ReactDOM from 'react-dom';
import ConnectedDetailList, { DetailList } from './DetailList';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import * as mocks from './../../test-helpers/mocks';

const mockStore = configureStore();

let initialState;
let store;
let wrapper;
const mockDeleteProduct = jest.fn();
const mockAddQuantity = jest.fn();
beforeEach(() => {
    jest.clearAllMocks();
    DetailList.prototype._slider = { addEventListener };

    initialState = {
        data: {
            orders: mocks.getOrders(),
            products: mocks.getProducts(),
            customers: mocks.getCustomers()
        }
    };
    //creates the store with any initial state or middleware needed  
    store = mockStore(initialState);
    // wrapper = mount(<Provider store={store}><ConnectedList /></Provider>);
    wrapper = shallow(<DetailList order={mocks.getOrders()[0]} item={mocks.getOrders()[0].items[0]} products={mocks.getProducts()} addQuantity={mockAddQuantity} deleteProduct={mockDeleteProduct} />);
});


xit('snapshot', () => {
    const tree = wrapper;
    expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Provider store={store}><ConnectedDetailList /></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
});


it('renders .... if no item is set', () => {
    wrapper = shallow(<DetailList order={mocks.getOrders()[0]} products={mocks.getProducts()} addQuantity={mockAddQuantity} deleteProduct={mockDeleteProduct} />);
    expect(wrapper.contains('....')).toBeTruthy();
});
it('renders .... if no products are loaded', () => {
    wrapper = shallow(<DetailList order={mocks.getOrders()[0]} products={[]} item={mocks.getOrders()[0].items[0]} addQuantity={mockAddQuantity} deleteProduct={mockDeleteProduct} />);
    expect(wrapper.contains('....')).toBeTruthy();
});
it('does not render .... if all is set', () => {
    expect(wrapper.contains('....')).toBeFalsy();
});
it('first round buttons trigger add quantity with -1', () => {
    wrapper.find('.round-button').at(0).simulate('click');
    expect(mockAddQuantity).toHaveBeenCalledWith(-1, 'B102', '1');
});
it('second round buttons trigger add quantity with 1', () => {
    wrapper.find('.round-button').at(1).simulate('click');
    expect(mockAddQuantity).toHaveBeenCalledWith(1, 'B102', '1');
});
it('deletePrompt is not visible initially', () => {
    expect(wrapper.find('.deletePrompt').hasClass('showDelete')).toBeFalsy();
});

it('delete-cross click shows deletePrompt', () => {
    wrapper.find('.delete-cross').simulate('click');
    expect(wrapper.find('.deletePrompt').hasClass('showDelete')).toBeTruthy();
});
it('no-button click hides deletePrompt', () => {
    wrapper.find('.delete-cross').simulate('click');
    wrapper.find('.no-button').simulate('click');

    expect(wrapper.find('.deletePrompt').hasClass('showDelete')).toBeFalsy();
});
it('yes-button click hides deletePrompt', () => {
    wrapper.find('.delete-cross').simulate('click');
    wrapper.find('.yes-button').simulate('click');

    expect(wrapper.find('.deletePrompt').hasClass('showDelete')).toBeTruthy();
});
it('yes-button click triggers deletefunction with product id and order id', () => {
    wrapper.find('.delete-cross').simulate('click');
    wrapper.find('.yes-button').simulate('click');

    expect(mockDeleteProduct).toHaveBeenCalledWith('B102','1');
});
