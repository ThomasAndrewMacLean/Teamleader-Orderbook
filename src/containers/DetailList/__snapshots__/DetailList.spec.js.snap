// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DetailList
    addQuantity={[Function]}
    deleteProduct={[Function]}
    item={
        Object {
            "product-id": "B102",
            "quantity": "10",
            "total": "49.90",
            "unit-price": "4.99",
          }
    }
    order={
        Object {
            "customer-id": "1",
            "id": "1",
            "items": Array [
              Object {
                "product-id": "B102",
                "quantity": "10",
                "total": "49.90",
                "unit-price": "4.99",
              },
            ],
            "total": "49.90",
          }
    }
    products={
        Array [
            Object {
              "category": "1",
              "description": "Screwdriver",
              "id": "A101",
              "price": "9.75",
            },
            Object {
              "category": "1",
              "description": "Electric screwdriver",
              "id": "A102",
              "price": "49.50",
            },
            Object {
              "category": "2",
              "description": "Basic on-off switch",
              "id": "B101",
              "price": "4.99",
            },
            Object {
              "category": "2",
              "description": "Press button",
              "id": "B102",
              "price": "4.99",
            },
            Object {
              "category": "2",
              "description": "Switch with motion detector",
              "id": "B103",
              "price": "12.95",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="deletePrompt"
>
          <div>
                    DELETE 
                    Press button
                    ?
          </div>
          <div
                    className="quatreCol resize-small"
          >
                    <button
                              className="big-button no-button"
                              onClick={[Function]}
                    >
                              NO
                    </button>
          </div>
          <div
                    className="quatreCol resize-small"
          >
                    <button
                              className="big-button yes-button"
                              onClick={[Function]}
                    >
                              YES
                    </button>
          </div>
</li>,
        <li
          className="list-item-detail"
>
          <div
                    className="quatreCol resize-small"
          >
                    B102
                    :
                    Press button
          </div>
          <div
                    className="quatreCol resize-small quantityAndButtons"
          >
                    <div
                              className="buttonwrap"
                    >
                              <button
                                        className="round-button"
                                        onClick={[Function]}
                              >
                                        -
                              </button>
                              <p
                                        className="pad"
                              >
                                        10
                              </p>
                              <button
                                        className="round-button"
                                        onClick={[Function]}
                              >
                                        +
                              </button>
                    </div>
          </div>
          <div
                    className="clear-small"
          >
                    <div
                              className="quatreCol resize-small"
                    >
                              <p
                                        className="pullright"
                              >
                                        €4.99
                              </p>
                    </div>
                    <div
                              className="quatreCol resize-small"
                    >
                              <p
                                        className="pullright"
                              >
                                        €49.90
                              </p>
                    </div>
          </div>
          <div
                    className="delete-cross"
                    onClick={[Function]}
          >
                    <Icon
                              color=""
                              icon="M192 1024h640l64-704h-768zM640 128v-128h-256v128h-320v192l64-64h768l64 64v-192h-320zM576 128h-128v-64h128v64z"
                              size={16}
                    />
          </div>
</li>,
      ],
      "className": "detail",
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              DELETE 
              Press button
              ?
</div>,
            <div
              className="quatreCol resize-small"
>
              <button
                            className="big-button no-button"
                            onClick={[Function]}
              >
                            NO
              </button>
</div>,
            <div
              className="quatreCol resize-small"
>
              <button
                            className="big-button yes-button"
                            onClick={[Function]}
              >
                            YES
              </button>
</div>,
          ],
          "className": "deletePrompt",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "DELETE ",
                "Press button",
                "?",
              ],
            },
            "ref": null,
            "rendered": Array [
              "DELETE ",
              "Press button",
              "?",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="big-button no-button"
                onClick={[Function]}
>
                NO
</button>,
              "className": "quatreCol resize-small",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NO",
                "className": "big-button no-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "NO",
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="big-button yes-button"
                onClick={[Function]}
>
                YES
</button>,
              "className": "quatreCol resize-small",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "YES",
                "className": "big-button yes-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "YES",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="quatreCol resize-small"
>
              B102
              :
              Press button
</div>,
            <div
              className="quatreCol resize-small quantityAndButtons"
>
              <div
                            className="buttonwrap"
              >
                            <button
                                          className="round-button"
                                          onClick={[Function]}
                            >
                                          -
                            </button>
                            <p
                                          className="pad"
                            >
                                          10
                            </p>
                            <button
                                          className="round-button"
                                          onClick={[Function]}
                            >
                                          +
                            </button>
              </div>
</div>,
            <div
              className="clear-small"
>
              <div
                            className="quatreCol resize-small"
              >
                            <p
                                          className="pullright"
                            >
                                          €4.99
                            </p>
              </div>
              <div
                            className="quatreCol resize-small"
              >
                            <p
                                          className="pullright"
                            >
                                          €49.90
                            </p>
              </div>
</div>,
            <div
              className="delete-cross"
              onClick={[Function]}
>
              <Icon
                            color=""
                            icon="M192 1024h640l64-704h-768zM640 128v-128h-256v128h-320v192l64-64h768l64 64v-192h-320zM576 128h-128v-64h128v64z"
                            size={16}
              />
</div>,
          ],
          "className": "list-item-detail",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "B102",
                ":",
                "Press button",
              ],
              "className": "quatreCol resize-small",
            },
            "ref": null,
            "rendered": Array [
              "B102",
              ":",
              "Press button",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="buttonwrap"
>
                <button
                                className="round-button"
                                onClick={[Function]}
                >
                                -
                </button>
                <p
                                className="pad"
                >
                                10
                </p>
                <button
                                className="round-button"
                                onClick={[Function]}
                >
                                +
                </button>
</div>,
              "className": "quatreCol resize-small quantityAndButtons",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="round-button"
                    onClick={[Function]}
>
                    -
</button>,
                  <p
                    className="pad"
>
                    10
</p>,
                  <button
                    className="round-button"
                    onClick={[Function]}
>
                    +
</button>,
                ],
                "className": "buttonwrap",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "-",
                    "className": "round-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "-",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "10",
                    "className": "pad",
                  },
                  "ref": null,
                  "rendered": "10",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "+",
                    "className": "round-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "+",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="quatreCol resize-small"
>
                  <p
                                    className="pullright"
                  >
                                    €4.99
                  </p>
</div>,
                <div
                  className="quatreCol resize-small"
>
                  <p
                                    className="pullright"
                  >
                                    €49.90
                  </p>
</div>,
              ],
              "className": "clear-small",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p
                    className="pullright"
>
                    €4.99
</p>,
                  "className": "quatreCol resize-small",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "€4.99",
                    "className": "pullright",
                  },
                  "ref": null,
                  "rendered": "€4.99",
                  "type": "p",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p
                    className="pullright"
>
                    €49.90
</p>,
                  "className": "quatreCol resize-small",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "€49.90",
                    "className": "pullright",
                  },
                  "ref": null,
                  "rendered": "€49.90",
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                color=""
                icon="M192 1024h640l64-704h-768zM640 128v-128h-256v128h-320v192l64-64h768l64 64v-192h-320zM576 128h-128v-64h128v64z"
                size={16}
/>,
              "className": "delete-cross",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "",
                "icon": "M192 1024h640l64-704h-768zM640 128v-128h-256v128h-320v192l64-64h768l64 64v-192h-320zM576 128h-128v-64h128v64z",
                "size": 16,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "li",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="deletePrompt"
>
            <div>
                        DELETE 
                        Press button
                        ?
            </div>
            <div
                        className="quatreCol resize-small"
            >
                        <button
                                    className="big-button no-button"
                                    onClick={[Function]}
                        >
                                    NO
                        </button>
            </div>
            <div
                        className="quatreCol resize-small"
            >
                        <button
                                    className="big-button yes-button"
                                    onClick={[Function]}
                        >
                                    YES
                        </button>
            </div>
</li>,
          <li
            className="list-item-detail"
>
            <div
                        className="quatreCol resize-small"
            >
                        B102
                        :
                        Press button
            </div>
            <div
                        className="quatreCol resize-small quantityAndButtons"
            >
                        <div
                                    className="buttonwrap"
                        >
                                    <button
                                                className="round-button"
                                                onClick={[Function]}
                                    >
                                                -
                                    </button>
                                    <p
                                                className="pad"
                                    >
                                                10
                                    </p>
                                    <button
                                                className="round-button"
                                                onClick={[Function]}
                                    >
                                                +
                                    </button>
                        </div>
            </div>
            <div
                        className="clear-small"
            >
                        <div
                                    className="quatreCol resize-small"
                        >
                                    <p
                                                className="pullright"
                                    >
                                                €4.99
                                    </p>
                        </div>
                        <div
                                    className="quatreCol resize-small"
                        >
                                    <p
                                                className="pullright"
                                    >
                                                €49.90
                                    </p>
                        </div>
            </div>
            <div
                        className="delete-cross"
                        onClick={[Function]}
            >
                        <Icon
                                    color=""
                                    icon="M192 1024h640l64-704h-768zM640 128v-128h-256v128h-320v192l64-64h768l64 64v-192h-320zM576 128h-128v-64h128v64z"
                                    size={16}
                        />
            </div>
</li>,
        ],
        "className": "detail",
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                DELETE 
                Press button
                ?
</div>,
              <div
                className="quatreCol resize-small"
>
                <button
                                className="big-button no-button"
                                onClick={[Function]}
                >
                                NO
                </button>
</div>,
              <div
                className="quatreCol resize-small"
>
                <button
                                className="big-button yes-button"
                                onClick={[Function]}
                >
                                YES
                </button>
</div>,
            ],
            "className": "deletePrompt",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "DELETE ",
                  "Press button",
                  "?",
                ],
              },
              "ref": null,
              "rendered": Array [
                "DELETE ",
                "Press button",
                "?",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="big-button no-button"
                  onClick={[Function]}
>
                  NO
</button>,
                "className": "quatreCol resize-small",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NO",
                  "className": "big-button no-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "NO",
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="big-button yes-button"
                  onClick={[Function]}
>
                  YES
</button>,
                "className": "quatreCol resize-small",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "YES",
                  "className": "big-button yes-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "YES",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="quatreCol resize-small"
>
                B102
                :
                Press button
</div>,
              <div
                className="quatreCol resize-small quantityAndButtons"
>
                <div
                                className="buttonwrap"
                >
                                <button
                                                className="round-button"
                                                onClick={[Function]}
                                >
                                                -
                                </button>
                                <p
                                                className="pad"
                                >
                                                10
                                </p>
                                <button
                                                className="round-button"
                                                onClick={[Function]}
                                >
                                                +
                                </button>
                </div>
</div>,
              <div
                className="clear-small"
>
                <div
                                className="quatreCol resize-small"
                >
                                <p
                                                className="pullright"
                                >
                                                €4.99
                                </p>
                </div>
                <div
                                className="quatreCol resize-small"
                >
                                <p
                                                className="pullright"
                                >
                                                €49.90
                                </p>
                </div>
</div>,
              <div
                className="delete-cross"
                onClick={[Function]}
>
                <Icon
                                color=""
                                icon="M192 1024h640l64-704h-768zM640 128v-128h-256v128h-320v192l64-64h768l64 64v-192h-320zM576 128h-128v-64h128v64z"
                                size={16}
                />
</div>,
            ],
            "className": "list-item-detail",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "B102",
                  ":",
                  "Press button",
                ],
                "className": "quatreCol resize-small",
              },
              "ref": null,
              "rendered": Array [
                "B102",
                ":",
                "Press button",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="buttonwrap"
>
                  <button
                                    className="round-button"
                                    onClick={[Function]}
                  >
                                    -
                  </button>
                  <p
                                    className="pad"
                  >
                                    10
                  </p>
                  <button
                                    className="round-button"
                                    onClick={[Function]}
                  >
                                    +
                  </button>
</div>,
                "className": "quatreCol resize-small quantityAndButtons",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="round-button"
                      onClick={[Function]}
>
                      -
</button>,
                    <p
                      className="pad"
>
                      10
</p>,
                    <button
                      className="round-button"
                      onClick={[Function]}
>
                      +
</button>,
                  ],
                  "className": "buttonwrap",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "-",
                      "className": "round-button",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "-",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "10",
                      "className": "pad",
                    },
                    "ref": null,
                    "rendered": "10",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "+",
                      "className": "round-button",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "+",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="quatreCol resize-small"
>
                    <p
                                        className="pullright"
                    >
                                        €4.99
                    </p>
</div>,
                  <div
                    className="quatreCol resize-small"
>
                    <p
                                        className="pullright"
                    >
                                        €49.90
                    </p>
</div>,
                ],
                "className": "clear-small",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p
                      className="pullright"
>
                      €4.99
</p>,
                    "className": "quatreCol resize-small",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "€4.99",
                      "className": "pullright",
                    },
                    "ref": null,
                    "rendered": "€4.99",
                    "type": "p",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p
                      className="pullright"
>
                      €49.90
</p>,
                    "className": "quatreCol resize-small",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "€49.90",
                      "className": "pullright",
                    },
                    "ref": null,
                    "rendered": "€49.90",
                    "type": "p",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  color=""
                  icon="M192 1024h640l64-704h-768zM640 128v-128h-256v128h-320v192l64-64h768l64 64v-192h-320zM576 128h-128v-64h128v64z"
                  size={16}
/>,
                "className": "delete-cross",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "",
                  "icon": "M192 1024h640l64-704h-768zM640 128v-128h-256v128h-320v192l64-64h768l64 64v-192h-320zM576 128h-128v-64h128v64z",
                  "size": 16,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "li",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
